cardano-config/src/Cardano/Node/Configuration/File/Consensus.hs

```diff
@@ -0,0 +1,55 @@
+-- | Options related to the Consensus layer
+module Cardano.Node.Configuration.File.Consensus
+  ( ConsensusConfiguration (..)
+  , GenesisConfigFlags (..)
+  ) where
+
+import Cardano.Ledger.BaseTypes
+import Cardano.Node.Configuration.Basics
+import Data.Aeson
+import Data.Default
+import Data.Time.Clock (DiffTime)
+import GHC.Generics (Generic)
+
+-- | In which mode should the node run.
+data ConsensusConfiguration
+  = PraosMode
+  | GenesisMode GenesisConfigFlags
+  deriving (Generic, Show)
+
+instance FromJSON ConsensusConfiguration where
+  parseJSON =
+    withObject "ConsensusMode" $ \v -> do
+      v .:? "ConsensusMode" >>= \case
+        Just "GenesisMode" -> GenesisMode <$> parseJSON (Object v)
+        Just "PraosMode" -> pure PraosMode
+        Nothing -> pure def
```

Comment:
I am not convinced that putting default values in JSON instances is a good idea. It conflates defaulting with deserialisation. I think those should be separated. Defaulting can be implemented like that: https://chrispenner.ca/posts/hkd-options#setting-default-values

Reply:
Good point. I have added some intermediate `Maybe`s to signal that some things still expect a default, and only when resolving the configuration at the end I pull in the default for those.